_posts/2018-12-18-blog-a-more-complete-front-end-testing-strategy.md

```diff
@@ -0,0 +1,249 @@
+
+---
+layout: post
+title:  "Explore a more complete front-end testing strategy"
+date:   2018-12-18 12:30:00 +0800
+categories: blog
+author: Michael Lin
+---
+
+
+![test-strategy](https://raw.githubusercontent.com/unadlib/integration-blog/testing_strategy/_site/assets/test-strategy.jpg)
+
+> The assumption in this article is that we are continuously developing a relatively large front-end project, and we have adopted a domain-driven design, as well as an object-oriented programming model. The front-end business logic is split into **`domain modules`**/**`UI components`**. So we may need to have a more complete testing strategy to assure such a front-end project.
+
+## General front-end testing
+
+Whether it's a traditional test model or a continuous delivery model, we typically define the following three types of tests:
+
+![generic-test](https://raw.githubusercontent.com/unadlib/integration-blog/testing_strategy/_site/assets/generic-test.png)
+
+- E2E
+
+End-to-end testing involves ensuring that the integrated components of an application function as expected. The entire application is tested in a real-world scenario such as communicating with the database, network, hardware and other applications.
+
+- IT
+
+Integration testing is a key software development life cycle strategy. Generally, small software systems are integrated and tested in a single phase, whereas larger systems involve several integration phases to build a complete system, such as integrating modules into low-level subsystems for integration with larger subsystems. Integration testing encompasses all aspects of a software system's performance, functionality and reliability.
+
+
+- UT
+
+Unit testing is a software testing method by which individual units of source code, sets of one or more computer program modules together with associated control data, usage procedures, and operating procedures, are tested to determine whether they are fit for use.
+
+
+
+## What's wrong with the general front-end testing?
+
+When we are evaluating the integrity of a test strategy, we try to validate it with some of the following criteria:
+
+* Pass acceptance criteria
+* Catch bugs as early as possible
+* Faster running speed, especially at the top level of testing
+* Find bugs efficiently
+* Test writing costs and maintenance costs
+* Continuous refactoring risks
+
+I think a better test strategy should be as common as possible with the above items.
```

Comment:
Do you mean there should be a test strategy where most things are done the same way? If so, I'd use consistent over common.

Reply:
I think my meaning is 'consistent'.